.gitlab/e2e/e2e.yml

```diff
@@ -124,21 +127,26 @@ k8s-e2e-otlp-main:
       annotations:
         - $EXTERNAL_LINKS_PATH
 
+.needs_new_e2e_template:
+  - go_e2e_deps
```

Comment:
These are usually defined in the main `gitlab-ci.yml`. The file layout could maybe be revised lately

Reply:
I wanted to have it as close to the `.new_e2e_template` block, but I can move it if you think that makes more sense

Reply:
I'm fine with it there. It was me saying it loud I would like to restructure the yaml of e2e 😄